src/Compilers/CSharp/Test/Semantic/Semantics/NameOfTests.cs

```diff
@@ -2373,5 +2366,497 @@ class Attr : System.Attribute { public Attr(string s) {} }";
+
+                var v = nameof(List<List<>>);
+                Console.WriteLine(v);
+                """).VerifyDiagnostics(
+                    // (4,21): error CS7003: Unexpected use of an unbound generic name
+                    // var v = nameof(List<List<>>);
+                    Diagnostic(ErrorCode.ERR_UnexpectedUnboundGenericName, "List<>").WithLocation(4, 21));
+        }
+
+        [Fact]
+        public void OpenTypeInNameof_NoNestedOpenTypes2()
+        {
+            CreateCompilation("""
+                using System;
+                using System.Collections.Generic;
+
+                var v = nameof(List<List<>[]>);
+                Console.WriteLine(v);
+                """).VerifyDiagnostics(
+                    // (4,21): error CS7003: Unexpected use of an unbound generic name
+                    // var v = nameof(List<List<>>);
+                    Diagnostic(ErrorCode.ERR_UnexpectedUnboundGenericName, "List<>").WithLocation(4, 21));
+        }
+
+        [Fact]
+        public void OpenTypeInNameof_NoNestedOpenTypes3()
+        {
+            CreateCompilation("""
+                using System;
+                using System.Collections.Generic;
+
+                var v = nameof(List<Outer<>.Inner>);
+                Console.WriteLine(v);
+                
+                public class Outer<T> { public class Inner { } }
+                """).VerifyDiagnostics(
+                    // (4,21): error CS7003: Unexpected use of an unbound generic name
+                    // var v = nameof(List<Outer<>.Inner>);
+                    Diagnostic(ErrorCode.ERR_UnexpectedUnboundGenericName, "Outer<>").WithLocation(4, 21));
+        }
+
+        [Fact]
+        public void OpenTypeInNameof_NoNestedOpenTypes4()
+        {
+            CreateCompilation("""
+                using System;
+                using System.Collections.Generic;
+
+                var v = nameof(List<Outer.Inner<>>);
+                Console.WriteLine(v);
+                
+                public class Outer { public class Inner<T> { } }
+                """).VerifyDiagnostics(
+                    // (4,27): error CS7003: Unexpected use of an unbound generic name
+                    // var v = nameof(List<Outer.Inner<>>);
+                    Diagnostic(ErrorCode.ERR_UnexpectedUnboundGenericName, "Inner<>").WithLocation(4, 27));
+        }
+
+        [Fact]
+        public void Nameof_NestedClosedType1()
+        {
+            CreateCompilation("""
+                using System;
+                using System.Collections.Generic;
+
+                var v = nameof(List<List<int>>);
+                Console.WriteLine(v);
+                """).VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void Nameof_NestedClosedType2()
+        {
+            CreateCompilation("""
+                using System;
+                using System.Collections.Generic;
+
+                var v = nameof(List<List<int>>);
```

Comment:
I couldn't see the difference with `Nameof_NestedClosedType1`

Reply:
fixed.